drivers/gpio/gpio_stm32.c

```diff
@@ -577,7 +654,17 @@ static int gpio_stm32_init(const struct device *device)
 
 	data->dev = device;
 
+#ifdef CONFIG_PM_DEVICE_IDLE
+	data->power_state = DEVICE_PM_LOW_POWER_STATE;
+	device_pm_enable(device);
+
+	return 0;
+#else
+#ifdef CONFIG_PM_DEVICE
+	data->power_state = DEVICE_PM_ACTIVE_STATE;
```

Comment:
omg, we also have `DEVICE_PM_STATE_ACTIVE`

Reply:
Yeah, it seems that `device->pm->fsm_state` (`DEVICE_PM_STATE_FOO`) and "Device Power Management States" (as used in `device_set_power_state` for instance are 2 different things....

Reply:
yep, got it, just wondering if we should not unify this. Btw, that is the reason I have no complains with the pr, because they are different things right now